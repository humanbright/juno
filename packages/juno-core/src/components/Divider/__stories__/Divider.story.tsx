import React, { ComponentProps } from 'react';

import { palette2, RcDivider, styled } from '@ringcentral/juno';
import {
  paletteChoice,
  switchToControlKnobs,
} from '@ringcentral/juno-storybook';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'ðŸš€ Cleanup Components/Divider',
  component: RcDivider,
  argTypes: {
    color: {
      options: paletteChoice,
      control: {
        type: 'select',
      },
    },
    titleBgColor: {
      options: paletteChoice,
      control: {
        type: 'select',
      },
    },
  },
} as Meta;

const P = styled.p`
  padding: 10px;
  margin: 0;
  color: ${palette2('interactive', 'f01')};
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

type DividerProps = ComponentProps<typeof RcDivider>;

export const Divider: Story<DividerProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return (
    <Wrapper
      style={{
        flexDirection: args.vertical ? 'row' : 'column',
        height: args.vertical ? '3em' : 'auto',
      }}
    >
      <P>test test test test test</P>
      <RcDivider {...args} />
      <P>test test test test test</P>
      <RcDivider {...args} />
      <P>test test test test test</P>
      <RcDivider {...args} />
    </Wrapper>
  );
};

Divider.args = {};

Divider.parameters = {
  tags: [
    {
      name: 'Spec',
      href: 'https://app.abstract.com/projects/d893ad32-0d8d-4e65-a37a-b4a5c881c51b/branches/master/commits/c098bb8a667d08ce4af62c20b580dbcf8629238e/files/8ffaf9bc-6939-4ef9-9bd1-6aa5b460e2ae/layers/9C67772C-84D8-432A-9B2A-A5CF971B0DAF',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

export const DividerExamples: Story<DividerProps> = () => {
  return (
    <>
      <Wrapper
        style={{
          flexDirection: 'column',
        }}
      >
        <P>test test test test test</P>
        <RcDivider title="divider title" />
        <P>test test test test test</P>
        <RcDivider title="divider title" />
        <P>test test test test test</P>
        <RcDivider title="divider title" />
      </Wrapper>
      <br />
      <Wrapper
        style={{
          flexDirection: 'row',
          height: '3em',
        }}
      >
        <P>test test test test test</P>
        <RcDivider vertical />
        <P>test test test test test</P>
        <RcDivider vertical />
        <P>test test test test test</P>
        <RcDivider vertical />
      </Wrapper>
    </>
  );
};

DividerExamples.argTypes = {
  color: {
    control: {
      type: null,
    },
  },
  titleBgColor: {
    control: {
      type: null,
    },
  },
};
