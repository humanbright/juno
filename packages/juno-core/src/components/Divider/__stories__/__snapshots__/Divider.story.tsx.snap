// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Divider Divider 1`] = `
.c2 {
  position: relative;
  margin: 0;
  border: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow: initial;
  width: 100%;
  height: 1px;
  background-color: #E5E5E5;
}

.c1 {
  padding: 10px;
  margin: 0;
  color: #066FAC;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
  style={
    Object {
      "flexDirection": "column",
      "height": "auto",
    }
  }
>
  <p
    className="c1"
  >
    test test test test test
  </p>
  <hr
    className="c2"
    role="separator"
  />
  <p
    className="c1"
  >
    test test test test test
  </p>
  <hr
    className="c2"
    role="separator"
  />
  <p
    className="c1"
  >
    test test test test test
  </p>
  <hr
    className="c2"
    role="separator"
  />
</div>
`;

exports[`Storyshots ðŸš€ Cleanup Components/Divider Divider Examples 1`] = `
Array [
  .c2 {
  position: relative;
  margin: 0;
  border: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow: initial;
  width: 100%;
  height: 1px;
  background-color: #E5E5E5;
}

.c2::after {
  content: 'divider title';
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #121212;
  position: absolute;
  left: 50%;
  top: 50%;
  padding: 0px 16px;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  background-color: #FFFFFF;
}

.c1 {
  padding: 10px;
  margin: 0;
  color: #066FAC;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    className="c0"
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
  </div>,
  <br />,
  .c2 {
  position: relative;
  margin: 0;
  border: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow: initial;
  width: 1px;
  height: 100%;
  background-color: #E5E5E5;
}

.c1 {
  padding: 10px;
  margin: 0;
  color: #066FAC;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    className="c0"
    style={
      Object {
        "flexDirection": "row",
        "height": "3em",
      }
    }
  >
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
    <p
      className="c1"
    >
      test test test test test
    </p>
    <hr
      className="c2"
      role="separator"
    />
  </div>,
]
`;
